### PR TITLE
Make usage of the performance.now method safer and more centralized

### LINKED ISSUE #2056: existence of window.performance does not imply performance.now method
Until [Android 4.4](http://caniuse.com/#search=performance), Android had a performance object but not the now object. I'm seeing reports of a wide range of niche browsers - Facebook, Chrome Mobile, Firefox - that have performance but not performance.now. Our feature checks for this need to be more robust.

### PR DESCRIPTION
**Pull request by tmcw**
Fixes #2056 

cc @ansis for the review

### CODE REVIEW

**Comment by ansis** [comment]
look good to me

File: js/util/browser/browser.js [hunk 1]
```diff
@@ -2,6 +2,27 @@
...
+var nowFn = (function() {
+    if (window.performance &&
+        window.performance.now) {
+        return window.performance.now.bind(window.performance);
+    } else {
+        return Date.now.bind(Date);
+    }
+}());
+
+module.exports.nowFn = nowFn;
```
Comment (lucaswoj) [inline js/util/browser/browser.js]: Can we call this method `now`? We don't need to use Hungarian notation here.

Comment (tmcw) [inline js/util/browser/browser.js]: We've been [calling its output now](https://github.com/mapbox/mapbox-gl-js/pull/2057/files#diff-e60957272552aeccdb45f65d82d523e1L47). Unless we rename the output, we shouldn't call this now.

Comment (lucaswoj) [inline js/util/browser/browser.js]: Something more descriptive, like `getEpochMilliseconds`, then?

**Comment by jfirebaugh** [comment]
Please call it `now`. There is nothing wrong with `var now = browser.now()`.

**Comment by tmcw** [comment]
Okay, renamed to `now`

**Pull request merged**